tests/Doctrine/Tests/ORM/Functional/Ticket/GH6823Test.php

```diff
@@ -0,0 +1,105 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\Tests\ORM\Functional\Ticket;
+
+use Doctrine\DBAL\Platforms\AbstractPlatform;
+use Doctrine\DBAL\Platforms\MySQLPlatform;
+use Doctrine\ORM\Mapping\Column;
+use Doctrine\ORM\Mapping\Entity;
+use Doctrine\ORM\Mapping\Id;
+use Doctrine\ORM\Mapping\ManyToOne;
+use Doctrine\ORM\Mapping\Table;
+use Doctrine\Tests\OrmFunctionalTestCase;
+
+use function method_exists;
+
+class GH6823Test extends OrmFunctionalTestCase
+{
+    public function testCharsetCollationWhenCreatingForeignRelations(): void
+    {
+        if (! $this->_em->getConnection()->getDatabasePlatform() instanceof MySQLPlatform) {
```

Comment:
@greg0ire one thing that crossed my mind, should we maybe change this test so that it runs on all platforms but only compares the sql for MySQL? That way we know it also works on Postgres

Reply:
If you can do that it would be great, yes 👍

Reply:
I think it requires a new test because on Postgres (and other dbs) those collations don't exist.

Or shall I just set those options after reading the metadata from the entity? That way we can have multiple setups per database?

Reply:
This sounds complex, and It's not a great idea to have complexity inside tests, so I'd say the first solution is best. I will let you judge if it is still worth it.

Reply:
I agree. Not gonna do it. I think it will work just fine.